sed/loader/flash/buffer_handler.py

```diff
@@ -71,6 +89,18 @@ def file_sets_to_process(self, force_recreate: bool = False) -> list[dict[str, P
             return self._file_paths
         return [file_set for file_set in self if any(not file_set[key].exists() for key in DF_TYP)]
 
+    def remove_invalid_files(self, config, h5_paths: list[Path]) -> list[Path]:
+        valid_h5_paths = []
+        for h5_path in h5_paths:
+            try:
+                dfc = DataFrameCreator(config_dataframe=config, h5_path=h5_path)
+                dfc.validate_channel_keys()
+                valid_h5_paths.append(h5_path)
```

Comment:
Yes. There are two reasons
- Because we don't process the files if they are already processed
- And in case I decide to always go through _save_buffer_file, removing files in a loop is not safe.

Another option is to open all h5 files and reuse those later. What do you think?

Reply:
It probably does not make a big difference.